packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -228,6 +227,21 @@ abstract class WidgetsBindingObserver {
   /// This method exposes notifications from [SystemChannels.lifecycle].
   void didChangeAppLifecycleState(AppLifecycleState state) { }
 
+  /// Called when a request is received from the system to exit the application.
+  ///
+  /// If any observer responds with [AppExitResponse.cancel], it will cancel the
+  /// exit. All observers will be asked before exiting.
+  ///
```

Comment:
Maybe worthwhile add document that there is no guarantee this is called before the app exists (see my comment on handleRequestAppExit as well)

Reply:
Done.